virttest/utils_hotplug.py

```diff
@@ -0,0 +1,401 @@
+                   'mtype': None, 'vcpus': None}
+    # Grab a dump of the guest - if we're using the --config,
+    # then get an --inactive dump.
+    extra_opts = ""
+    if "--config" in options or vm.is_dead():
+        extra_opts = "--inactive"
+    vm_xml = libvirt_xml.VMXML.new_from_dumpxml(vm.name, extra_opts)
+    cpu_xmldata['mtype'] = vm_xml.os.machine
+    try:
+        cpu_xmldata['current_vcpu'] = int(vm_xml.current_vcpu)
+    except LibvirtXMLNotFoundError:
+        logging.debug("current vcpu value not present in xml")
+    cpu_xmldata['vcpu'] = int(vm_xml.vcpu)
+    return cpu_xmldata
+
+
+def hotplug_supported(vm_name, mtype):
+    """
+    hotplug support check for ppc64le
+
+    :param vm_name: VM name
+    :param mtype: machine type
+
+    :return: True if supported and False if not
+    """
+    supported = False
+    if "ppc64" in platform.machine():
+        cmd = '{\"execute\":\"query-machines\"}'
+        json_result = virsh.qemu_monitor_command(vm_name, cmd, "--pretty",
+                                                 debug=False)
+        result = json.loads(json_result.stdout)
+        for item in result['return']:
+            if item['name'] == mtype:
+                if item['hotpluggable-cpus'] == 'True':
+                    supported = True
+    else:
+        # For now returning for other arch by default true
+        supported = True
+    return supported
+
+
+def affinity_from_vcpuinfo(vm):
+    """
+    Returns list of the vcpu's affinity from
+    virsh vcpuinfo output
+
+    :param vm_name: VM Name
+
+    :return: affinity list of VM
+    """
+    output = virsh.vcpuinfo(vm.name).stdout.rstrip()
+    affinity = re.findall('CPU Affinity: +[-y]+', output)
+    total_affinity = [list(vcpu_affinity.split()[-1].strip())
+                      for vcpu_affinity in affinity]
+    return total_affinity
+
+
+def affinity_from_xml(vm):
+    """
+    Returns dict of the vcpu's affinity from
+    guest xml
+
+    :param: VM object
+
+    :return: dict of affinity of VM
+    """
+    host_cpu_count = utils.count_cpus()
+    xml_affinity_list = []
+    xml_affinity = {}
+    try:
+        vmxml = libvirt_xml.VMXML.new_from_dumpxml(vm.name)
+        xml_affinity_list = vmxml['cputune'].vcpupins
+    except LibvirtXMLNotFoundError:
+        logging.debug("No <cputune> element find in domain xml")
+        return xml_affinity
+    # Store xml_affinity_list to a dict
+    for vcpu in xml_affinity_list:
+        xml_affinity[vcpu['vcpu']] = "".join(
+            libvirt.cpus_string_to_affinity_list(vcpu['cpuset'],
+                                                 host_cpu_count))
+    return xml_affinity
+
+
+def affinity_from_vcpupin(vm):
+    """
+    Returns dict of vcpu's affinity from virsh vcpupin output
+
+    :param: VM object
+
+    :return: dict of affinity of VM
+    """
+    vcpupin_output = []
+    vcpupin_affinity = {}
+    host_cpu_count = utils.count_cpus()
```

Comment:
autotest dependency can be removed,
`host_cpu_count = len(utils_misc.get_cpu_processors())`